### PR TITLE
Fix-misleading-prints-for-headers-pool-info

### PR DESCRIPTION
**Pull request by SebastianMarian**
* Fixed missleading prints for headers pool info
* Fixed value of maxNumHeadersToKeepPerShard from blockTracker
* Fixed marshal problems for empty slices instead nil in bootstrapStorer when protobuf is used
* Removed unnecessary code

### CODE REVIEW

File: process/block/metablock.go [hunk 1]
```diff
@@ -1140,17 +1150,33 @@ func (mp *metaProcessor) CommitBlock(
...
+		log.Trace("pools info",
+			"shard", shardID,
+			"num headers", mp.dataPool.Headers().GetNumHeaders(shardID))
+	}
+
+	log.Trace("pools info",
+		"shard", core.MetachainShardId,
+		"num headers", mp.dataPool.Headers().GetNumHeaders(core.MetachainShardId))
+
+	numShardsToKeepHeaders := int(mp.shardCoordinator.NumberOfShards()) + 1
```
Comment (andreibancioiu) [inline process/block/metablock.go]: Please add a comment why +1 (metachain)?

File: process/block/metablock.go [hunk 2]
```diff
@@ -1140,17 +1150,33 @@ func (mp *metaProcessor) CommitBlock(
...
+	log.Debug("pools info",
+		"total headers", mp.dataPool.Headers().Len(),
+		"headers pool capacity", mp.dataPool.Headers().MaxSize() * numShardsToKeepHeaders,
```
Comment (andreibancioiu) [inline process/block/metablock.go]: Optional: extract new variable, `capacity`.

File: process/block/shardblock.go [hunk 3]
```diff
@@ -904,19 +904,33 @@ func (sp *shardProcessor) CommitBlock(
...
+func (sp *shardProcessor) displayPoolsInfo() {
+	log.Trace("pools info",
+		"shard", sp.shardCoordinator.SelfId(),
+		"num headers", sp.dataPool.Headers().GetNumHeaders(sp.shardCoordinator.SelfId()))
+
+	log.Trace("pools info",
+		"shard", core.MetachainShardId,
+		"num headers", sp.dataPool.Headers().GetNumHeaders(core.MetachainShardId))
+
+	numShardsToKeepHeaders := 2
```
Comment (andreibancioiu) [inline process/block/shardblock.go]: Comment why 2? Self + metachain?

File: process/block/metablock.go (hunk 1 above)
Comment (SebastianMarian) [reply process/block/metablock.go]: Yes. Added comment

File: process/block/shardblock.go (hunk 3 above)
Comment (SebastianMarian) [reply process/block/shardblock.go]: Yes. Added comment

File: process/block/metablock.go (hunk 2 above)
Comment (SebastianMarian) [reply process/block/metablock.go]: Done

File: process/block/metablock.go [hunk 4]
```diff
@@ -1879,9 +1907,5 @@ func (mp *metaProcessor) getPendingMiniBlocks() []bootstrapStorage.PendingMiniBl
 		}
 	}
```
Comment (AdoAdoAdo) [inline process/block/metablock.go]: why is this removed? 
are we sure that we can never return empty pendingMiniBocksInfo?
if these are afterwards serialized, there might be some issues
